packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -696,13 +730,302 @@ describe("Garbage Collection Tests", () => {
+				const nodeData: IGarbageCollectionNodeData = {
+					outboundRoutes: [],
+					unreferencedTimestampMs: 123,
+				};
+				gcState.gcNodes[nodes[0]] = nodeData;
+
+				// Create a tombstone blob a node as tombstone. This will be returned when the garbage collector asks for
+				// the tombstone blob.
+				const gcTombstoneBlobId = "tombstone";
+				gcSnapshotTree.blobs[gcTombstoneBlobKey] = gcTombstoneBlobId;
+				const tombstones = [nodes[0]];
+
+				// Create a tombstone blob a node as tombstone. This will be returned when the garbage collector asks for
+				// the tombstone blob.
+				const gcDeletedBlobId = "deletedNodes";
+				gcSnapshotTree.blobs[gcDeletedBlobKey] = gcDeletedBlobId;
+				const deletedBlobs = [nodes[0]];
+
+				// Create a base snapshot that contains the GC snapshot tree.
+				const baseSnapshot = getDummySnapshotTree();
+				baseSnapshot.trees[gcTreeKey] = gcSnapshotTree;
+
+				const metadataBlobId = "metadata";
+				const metadata: IContainerRuntimeMetadata = {
+					gcFeature: gcVersion,
+					summaryFormatVersion: 1,
+					message: undefined,
+				};
+				baseSnapshot.blobs[metadataBlobName] = metadataBlobId;
+
+				const gcBlobMap: Map<string, any> = new Map();
+				gcBlobMap.set(gcBlobId, gcState);
+				gcBlobMap.set(gcTombstoneBlobId, tombstones);
+				gcBlobMap.set(gcDeletedBlobId, deletedBlobs);
+				gcBlobMap.set(metadataBlobId, metadata);
+
+				return { baseSnapshot, gcBlobMap };
+			}
+
+			it("reads all GC data from base snapshot when GC version does not change", async () => {
+				const { baseSnapshot, gcBlobMap } = getSnapshotWithGCVersion(currentGCVersion);
+				const garbageCollector = createGCOverride(
+					baseSnapshot,
+					gcBlobMap,
+					currentGCVersion,
+				);
+
+				// GC state, tombstone state and deleted nodes should all be read from base snapshot.
+				const baseSnapshotData = await garbageCollector.baseSnapshotDataP;
+				assert(
+					baseSnapshotData !== undefined,
+					"base snapshot was not initialized correctly",
+				);
+				assert(
+					baseSnapshotData.gcState !== undefined,
+					"GC state in base snapshot should not be available",
+				);
+				assert(
+					baseSnapshotData.tombstones !== undefined,
+					"Tombstone state in base snapshot should be available",
+				);
+				assert(
+					baseSnapshotData.deletedNodes !== undefined,
+					"Deleted nodes in base snapshot should be available",
+				);
+
+				// Initialize from the base state and validate that tombstones and deleted state both have one entry
+				// as per the base snapshot.
+				await garbageCollector.initializeBaseState();
+				assert.strictEqual(garbageCollector.tombstones.length, 1);
+				assert.strictEqual(garbageCollector.deletedNodes.size, 1);
+			});
+
+			it("discards GC state and tombstone state in base snapshot when GC version changes", async () => {
+				const updatedGCVersion = currentGCVersion + 1;
+				const { baseSnapshot, gcBlobMap } = getSnapshotWithGCVersion(updatedGCVersion);
```

Comment:
Do you want to move this to within `createGCOverride`, so then you just pass the gcVersion and it creates the snapshot/blob map and creates the GC?  Otherwise you could accidentally pass different gcVersions for the snapshot contents and the metadata.

Reply:
That's a good point.